src/cpp/session/SessionAsyncRProcess.cpp

```diff
@@ -220,7 +235,27 @@ void AsyncRProcess::onStderr(const std::string& output)
 
 bool AsyncRProcess::onContinue()
 {
-   return !terminationRequested_;
+   if (terminationRequested_)
+      return false;
+   
+   // check for request requiring a response
+   if (ipcRequests_.exists())
+   {
+      core::Error error = r::exec::RFunction(".rs.rstudioapi.processRequest")
+            .addParam(ipcRequests_.absolutePathNative())
+            .addParam(ipcResponse_.absolutePathNative())
+            .addParam(sharedSecret_)
+            .call();
+
+      if (error)
+      {
+         LOG_ERROR(error);
+         ipcRequests_.removeIfExists();
```

Comment:
This returns an error that should be logged, so that if for some reason we get stuck processing the file over and over because we can't delete it we'll have a bunch of log entries as a clue. If we wanted to be extra paranoid we could add a flag that tells us we've already processed the request (so the file can be ignored even if we fail to delete it), but that's probably overkill.